File: Day1/python/main.py
import numpy as np

def main():
    with open("C:/Users/Paul/PycharmProjects/AdventOfCode/Day1/data.txt") as f:
        lines = [line.rstrip('\n') for line in f]
    result = []
    current = 0
    for line in lines:
        if line != "":
            current += int(line)
        else:
            result.append(current)
            current = 0
    print("Solution of the first part is: " + str(np.max(result)))
    
    ind = np.argpartition(result, -3)[-3:]
    sum = 0
    for index in ind:
        sum += result[index]
    print("Solution of the second part is: " + str(sum))
        

if __name__ == "__main__":
    main()

File: Day1/rust/src/main.rs
use std::{fs};

fn main() {
    let contents = fs::read_to_string("C:/Users/Paul/PycharmProjects/AdventOfCode/Day1/data.txt")
        .expect("Wrong file path!");
    let data: Vec<_> = contents.lines().collect();

    let mut result1: Vec<i32> = Vec::new();
    let mut sum_of_elve = 0;

    for line in data.iter() {
        match line {
           &"" => {
            result1.push(sum_of_elve);
            sum_of_elve = 0;
           },
           _ => {
            let line_as_int: i32 = line.parse().unwrap();
            sum_of_elve = sum_of_elve + line_as_int;
           },
        }
    }
    let max_value = result1.iter().max();
    match max_value {
        Some(max) => println!( "Max value: {}", max),
        None      => println!( "Vector is empty" ),
    }

    
}
